wheatley/main.py

```diff
@@ -44,46 +45,125 @@ def create_row_generator(args):
-    # row_gen = DixonoidsGenerator(6, DixonoidsGenerator.DixonsRules)
-    # row_gen = PlaceNotationGenerator.stedman(11)
-
     return row_gen
 
 
-def create_rhythm(args):
+def create_rhythm(peal_speed, inertia, max_bells_in_dataset, handstroke_gap, use_wait):
     """ Generates a rhythm object according to the given CLI arguments. """
-
-    try:
-        peal_speed = parse_peal_speed(args.peal_speed)
-    except PealSpeedParseError as e:
-        sys.exit(str(e))
-
     regression = RegressionRhythm(
-        args.inertia,
-        handstroke_gap=args.handstroke_gap,
+        inertia,
+        handstroke_gap=handstroke_gap,
         peal_speed=peal_speed,
-        max_bells_in_dataset=args.max_bells_in_dataset
+        max_bells_in_dataset=max_bells_in_dataset
     )
 
-    if not args.keep_going:
+    if use_wait:
         return WaitForUserRhythm(regression)
 
     return regression
 
 
+def get_version_number():
+    """
+    Try to read the file with the version number, if not print an error and set a poison value as the
+    version.
+    """
+    try:
+        version_file_path = os.path.join(os.path.split(__file__)[0], "version.txt")
+
+        with open(version_file_path) as f:
+            return f.read()
+    except IOError:
+        return "<NO VERSION FILE FOUND>"
+
+
 def configure_logging():
     """ Sets up the logging for the bot. """
-
     logging.basicConfig(level=logging.WARNING)
 
+    logging.getLogger(Bot.logger_name).setLevel(logging.INFO)
     logging.getLogger(RingingRoomTower.logger_name).setLevel(logging.INFO)
     logging.getLogger(RowGenerator.logger_name).setLevel(logging.INFO)
     logging.getLogger(RegressionRhythm.logger_name).setLevel(logging.INFO)
     logging.getLogger(WaitForUserRhythm.logger_name).setLevel(logging.INFO)
 
 
+def server_main(override_args=None, stop_on_join_tower=False):
+    """
+    The main function of Wheatley when spawned by the Ringing Room server.
+    This has many many fewer options than the standard `main` function, on the basis that this running mode
+    is designed to take its parameters over SocketIO whilst running, rather than from the CLI args.
+    """
+    __version__ = get_version_number()
+
+    parser = argparse.ArgumentParser(
+        description="A bot to fill in bells during ringingroom.com practices (server mode)."
+    )
+
+    parser.add_argument(
+        "room_id",
+        type=int,
+        help="The numerical ID of the tower to join, represented as a row on 9 bells, \
+              e.g. 763451928."
+    )
+    parser.add_argument(
+        "-p", "--port",
+        type=int,
+        help="The port of the SocketIO server (which must be hosted on localhost)."
+    )
+    parser.add_argument(
+        "-l", "--look-to-time",
+        type=float,
+        help="Set to the time when 'Look to' was called if Wheatley was spawned because 'look to' was \
+              called and Wheatley is needed."
+    )
+
+    # Misc arguments
```

Comment:
Would it be more usable to have it log the version to `debug`, so it can be seen in the logs on ringing room by changing the logging level?

Reply:
Oh yes good idea!

Reply:
Done